GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -449,6 +450,27 @@ private void GetBranchRevisionFilter(ArgumentBuilder filter)
                     filter.Add(isSimpleBranchFilter ? BranchFilter : $"--branches={BranchFilter}");
                 }
             }
+            else if (IsShowCurrentBranchOnlyChecked)
+            {
+                // Git default, no option added except for stashes
+
+                if (AppSettings.ShowStashes && !string.IsNullOrWhiteSpace(currentBranch.Value))
+                {
+                    addFirstStashRef();
+
+                    // current branch must be an option due to --glob
+                    filter.Add(currentBranch.Value);
+                }
+            }
+            else if (IsShowFilteredBranchesChecked && !string.IsNullOrWhiteSpace(BranchFilter))
+            {
+                addFirstStashRef();
+
+                // Show filtered branches only
+                // Do not add suffix /* to the filter
+                bool isSimpleBranchFilter = BranchFilter.IndexOfAny(new[] { '?', '*', '[' }) == -1;
+                filter.Add(isSimpleBranchFilter ? BranchFilter : $"--branches={BranchFilter}");
```

Comment:
This is too detailed for the Rule Of Three IMO, i.e factor out, please.

Reply:
See above...